lib/blemulator/simulation/simulated_characteristic.dart

```diff
@@ -34,7 +34,10 @@ class SimulatedCharacteristic {
   Future<Uint8List> read() async =>
       isReadable ? _value : Future.error("This characteristic is not readable");
 
-  void write(Uint8List value) {
+  Future<void> write(Uint8List value) async {
+    if (!isWritableWithResponse || !isWritableWithoutResponse) {
+      return Future.error("This characteristic is not writeable");
```

Comment:
Perhaps the checks should be done by simulation manager in error_mixin or whatever it was called? Either way: create a proper BleError here, don't throw a string.

Reply:
1. We decided to throw these errors from characteristic_mixin because now user loses this error handling when he forget to call super.write() in characteristic's overridden "write" method.
2. I will also add proper BleError for that case instead of returning raw string
3. We don't simulate writing with or without response. I created a separate issue for that (https://github.com/Polidea/FlutterBleLib/issues/234)